src/lib/coreConnection.ts

```diff
@@ -349,31 +348,32 @@ export class CoreConnection extends EventEmitter {
 	/**
 	 * Like a subscribe, but automatically renews it upon reconnection
 	 */
-	autoSubscribe (publicationName: string, ...params: Array<any>): Promise<string> {
-		return this.subscribe(publicationName, ...params)
-		.then((subscriptionId: string) => {
-			this._autoSubscriptions[subscriptionId] = {
-				publicationName: publicationName,
-				params: params
-			}
-			return subscriptionId
-		})
+	async autoSubscribe (publicationName: string, ...params: Array<any>): Promise<string> {
+		const subscriptionId = await this.subscribe(publicationName, ...params)
+		this._autoSubscriptions[subscriptionId] = {
+			publicationName: publicationName,
+			params: params
+		}
+		return subscriptionId
 	}
 	unsubscribe (subscriptionId: string): void {
-		this.ddp.ddpClient.unsubscribe(subscriptionId)
+		this.ddp.ddpClient && this.ddp.ddpClient.unsubscribe(subscriptionId)
 		delete this._autoSubscriptions[subscriptionId]
 	}
 	observe (collectionName: string): Observer {
+		if (!this.ddp.ddpClient) {
+			throw new Error('observe: DDP client not initialised')
+		}
 		return this.ddp.ddpClient.observe(collectionName)
 	}
 	getCurrentTime (): number {
-		return this._timeSync.currentTime()
+		return this._timeSync && this._timeSync.currentTime() || 0
 	}
 	hasSyncedTime (): boolean {
-		return this._timeSync.isGood()
+		return this._timeSync && this._timeSync.isGood() || false
 	}
 	syncTimeQuality (): number | null {
-		return this._timeSync.quality
+		return this._timeSync && this._timeSync.quality || null
```

Comment:
should 0 be a valid number here? In other words should this be `||` or `??`

Reply:
Resolved in nytamin commit https://github.com/nrkno/tv-automation-server-core-integration/pull/34/commits/470c937f6f3027241ac5d13ccca71228e00c6c19

Reply:
Not quite, he introduced the `?.` operator, but didnt resolve my question about `||` vs `??`

Reply:
Resolved by @nytamin in commit https://github.com/nrkno/tv-automation-server-core-integration/pull/34/commits/470c937f6f3027241ac5d13ccca71228e00c6c19